examples/mnist/eth_ngram_mnist.py

```diff
@@ -0,0 +1,171 @@
+        accuracy['proportion'].append(100 * torch.sum(labels[i - update_interval:i].long() \
+                                                == proportion_pred) / update_interval)
+        accuracy['ngram'].append(100 * torch.sum(labels[i - update_interval:i].long() \
+                                                == ngram_pred) / update_interval)
+
+        for eval in ['all_activity', 'proportion', 'ngram']:
+            print(f"{eval} accuracy: {accuracy[eval][-1]} (last), {np.mean(accuracy[eval])} (average), {np.max(accuracy[eval])} (best)")
+
+        # Assign labels to excitatory layer neurons.
+        assignments, proportions, rates = assign_labels(spike_record, labels[i - update_interval:i], 10, rates)
+
+    # Get next input sample.
+    sample = next(data_loader)
+    inpts = {'X' : sample}
+
+    # Run the network on the input.
+    network.run(inpts=inpts, time=time)
+
+    # Get voltage recording.
+    exc_voltages = exc_voltage_monitor.get('v')
+    inh_voltages = inh_voltage_monitor.get('v')
+
+    # Add to spikes recording.
+    spike_record[i % update_interval] = spikes['Ae'].get('s').t()
+
+    # Optionally plot various simulation information.
+    if plot:
+        inpt = inpts['X'].view(time, 784).sum(0).view(28, 28)
+        input_exc_weights = network.connections[('X', 'Ae')].w
+        square_weights = get_square_weights(input_exc_weights.view(784, n_neurons), n_sqrt, 28)
+        square_assignments = get_square_assignments(assignments, n_sqrt)
+        voltages = {'Ae' : exc_voltages, 'Ai' : inh_voltages}
+
+        if i == 0:
+            inpt_axes, inpt_ims = plot_input(images[i].view(28, 28), inpt, label=labels[i])
+            spike_ims, spike_axes = plot_spikes({layer : spikes[layer].get('s') for layer in spikes})
+            weights_im = plot_weights(square_weights)
+            assigns_im = plot_assignments(square_assignments)
+            perf_ax = plot_performance(accuracy)
+            voltage_ims, voltage_axes = plot_voltages(voltages)
+
+        else:
+            inpt_axes, inpt_ims = plot_input(images[i].view(28, 28), inpt, label=labels[i], axes=inpt_axes, ims=inpt_ims)
+            spike_ims, spike_axes = plot_spikes({layer : spikes[layer].get('s') for layer in spikes},
+                                                ims=spike_ims, axes=spike_axes)
+            weights_im = plot_weights(square_weights, im=weights_im)
+            assigns_im = plot_assignments(square_assignments, im=assigns_im)
+            perf_ax = plot_performance(accuracy, ax=perf_ax)
+            voltage_ims, voltage_axes = plot_voltages(voltages, ims=voltage_ims, axes=voltage_axes)
+
+        plt.pause(1e-8)
+
+    network._reset()  # Reset state variables.
+
+print('Progress: %d / %d (%.4f seconds)\n' % (n_train, n_train, t() - start))
+print('Training complete.\n')
+
+print ('Saving necessary items.\n')
+
+with open('/mnt/nfs/work1/rkozma/hqkhan/output.txt', 'w') as out:
```

Comment:
Is there any reason you're saving things here, rather than re-using the results-saving code in the `bindsnet_experiments` repo?